devices/flash-nrf9160/flashdrv.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * NRF9160 Flash driver
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+/* Copy of flash-stm32 code */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include <devices/devs.h>
+
+/* number of flash devices */
+#define FLASH_NO 1
+
+/* clang-format off */
+enum {nvmc_ready = 256, nvmc_config = 321, nvmc_eraseall = 323, nvmc_configns = 353};
+/* clang-format on */
+
+static const struct {
+	u32 start;
+	u32 end;
+} flashParams[FLASH_NO] = {
+	{ FLASH_PROGRAM_1_ADDR, FLASH_PROGRAM_1_ADDR + FLASH_PROGRAM_BANK_SIZE }
+};
+
+
+static int flashdrv_isValidAddress(unsigned int minor, u32 off, size_t size)
+{
+	size_t fsize = flashParams[minor].end - flashParams[minor].start;
+
+	if (off < fsize && (off + size) <= fsize)
+		return 1;
```

Comment:
We aspire to produce all new code according to new rules, applies to the whole PR
```suggestion
	if (off < fsize && (off + size) <= fsize) {
		return 1;
	}
```

Reply:
Yeah, seems to be reasonable